src/toast/ops/hwpss_model.py

```diff
@@ -0,0 +1,936 @@
+
+        return local_good, central_mag
+
+    def _average_magnitude(self, dets, coeff):
+        mag = dict()
+        if self.time_drift:
+            # 4 values per harmonic, 2f is index 1
+            re_comp = 4 * 1 + 0
+            im_comp = 4 * 1 + 2
+        else:
+            # 2 values per harmonic, 2f is index 1
+            re_comp = 2 * 1 + 0
+            im_comp = 2 * 1 + 1
+        n_chunk = coeff.shape[2]
+        for idet, det in enumerate(dets):
+            ch_mag = list()
+            for ch in range(n_chunk):
+                ch_mag.append(
+                    np.sqrt(
+                        coeff[idet, re_comp, ch] ** 2 + coeff[idet, im_comp, ch] ** 2
+                    )
+                )
+            mag[det] = np.mean(ch_mag)
+        return mag
+
+    def _fit_chunk(
+        self, obs, dets, indx, start, end, sincos, sh_flags, det_flags, reltime, coeff
+    ):
+        log = Logger.get()
+        ch_timer = Timer()
+        ch_timer.start()
+
+        # The sample slice
+        slc = slice(start, end, 1)
+        slc_samps = end - start
+
+        if reltime is None:
+            ch_reltime = None
+        else:
+            ch_reltime = reltime[slc]
+
+        obs_cov = hwpss_compute_coeff_covariance(
+            sincos[slc],
+            sh_flags[slc],
+            comm=obs.comm.comm_group,
+            times=ch_reltime,
+            time_drift=self.time_drift,
+        )
+        if obs_cov is None:
+            msg = f"HWPSS Model {obs.name}[{indx}] ({slc_samps} samples)"
+            msg += " failed to compute coefficient"
+            msg += " covariance.  Change chunking or number of harmonics."
+            raise RuntimeError(msg)
+
+        msg = f"HWPSS Model {obs.name}[{indx}]: built coefficient covariance in"
+        log.verbose_rank(msg, comm=obs.comm.comm_group, timer=ch_timer)
+
+        for idet, det in enumerate(dets):
+            good_samp = det_flags[det][slc] == 0
+            if np.count_nonzero(good_samp) < coeff.shape[1]:
+                # Not very many good samples, set coefficients to zero
+                msg = f"HWPSS Model {obs.name}[{indx}] {det}: insufficient good "
+                msg += "samples, setting coefficients to zero"
+                log.verbose(msg)
+                coeff[idet, :, indx] = 0
+                continue
+            sig = np.array(obs.detdata[self.det_data][det, slc])
+            dc = np.mean(sig[good_samp])
+            sig -= dc
+
+            cf = hwpss_compute_coeff(
+                sincos[slc],
+                sig,
+                det_flags[det][slc],
+                obs_cov[0],
+                obs_cov[1],
+                times=ch_reltime,
+                time_drift=self.time_drift,
+            )
+            if idet == 0:
+                cfstr = ""
+                for ic in cf:
+                    cfstr += f"{ic} "
+                # print(f"Det Coeff {det}[{indx}] = {cfstr}")
```

Comment:
Maybe time to remove this and other debug statements?

Reply:
Done.